docs/configuration.md

```diff
@@ -0,0 +1,165 @@
+    </tr>
+    <tr>
+      <td>type</td>
+      <td><code>'validation' | 'style'`</code></td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>summary</td>
+      <td><code>string</code></td>
+      <td>A short summary of the rule and its intended purpose</td>
+    </tr>
+    <tr>
+      <td>description</td>
+      <td><code>string</code></td>
+      <td>A long-form description of the rule formatted in markdown</td>
+    </tr>
+    <tr>
+      <td>severity</td>
+      <td><code>Error = 0, Warning = 1, Information = 2, Hint = 3</code></td>
+      <td>The severity of results this rule generates</td>
+    </tr>
+    <tr>
+      <td>tags</td>
+      <td><code>string[]</code></td>
+      <td>Tags attached to the rule, which can be used for organizational purposes</td>
+    </tr>
+    <tr>
+      <td>enabled</td>
+      <td><code>boolean</code></td>
+      <td>should the rule be enabled by default?</td>
+    </tr>        
+    <tr>
+      <td>when</td>
+      <td><code><a href="#when">When</a></code></td>
+      <td>A filter object to narrow down the selected items.</td>
+    </tr>
+  </tbody>
+</table>
+
+### Then
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>    
+    <tr>
+      <td>function</td>
+      <td><code>string</code></td>
+      <td><b>Required</b> Name of the function to run. Note: to use a function you must first register it (<code>spectral.addFunctions(...)</code>). Please check out the default common functions.</td>
+    </tr>
+    <tr>
+      <td>field</td>
+      <td><code>string</code></td>
+      <td>Name of the field to narrow by. Given this field, the function will receive value of that field.</td>
+    </tr>    
+    <tr>
+      <td>functionOptions</td>
+      <td><code>Object</code></td>
+      <td>Options passed to the function. Specific to the function in use.</td>
+    </tr>
+  </tbody>
+</table>
+
+### When
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>    
+    <tr>
+      <td>field</td>
+      <td><code>string</code></td>
+      <td><b>Required</b> The <code>path.to.prop</code> to field, or special <code>@key</code> value to target keys for matched <code>given</code> object. <br/><br/> Example: if the target object is an oas object and given = <code>$..responses[*]</code>, then <code>@key</code> would be the response code (200, 400, etc)</td>
+    </tr>
+    <tr>
+      <td>patterns</td>
+      <td><code>string</code></td>
+      <td>A regex pattern</td>
+    </tr>
+  </tbody>
+</table>
+
+## Configuring rulesets via CLI
+
+```bash
+spectral lint foo.yaml --ruleset=path/to/ruleset.yaml
+```
+
+### Ruleset validation
+
+We use JSON Schema & AJV to validate your rulesets file and help you spot issues early.
+
+**Example output**
+
+```bash
+spectral lint some-oas.yaml --ruleset invalid-ruleset.json
```

Comment:
This example could be “acme-company.yaml” to make it clear what it’s for. Rulesets for rulesets sake won’t excite people.

Reply:
lol yeah good idea :)